pylint/config.py

```diff
@@ -795,8 +813,18 @@ def options_by_section(self):
     def options_and_values(self, options=None):
         if options is None:
             options = self.options
-        for optname, optdict in options:
-            yield (optname, optdict, self.option_value(optname))
+        options_dict = dict(options)
+
+        for option_name, option_dict in options:
+            if option_name not in self.config.value_set_order:
+                yield (option_name, option_dict, self.option_value(option_name))
+
+        for option_name in self.config.value_set_order:
+            try:
+                yield (option_name, options_dict[option_name],
+                       self.option_value(option_name))
+            except KeyError:
```

Comment:
I need to double check why this will raise again. I think it was related to how regex is handled by optparse, but I can't remember